File: 1556-make-two-arrays-equal-by-reversing-subarrays/1556-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        unordered_map<int, int> map;
        for (int i = 0; i < arr.size(); ++i) {
            map[arr[i]]++;
            map[target[i]]--;
        }
        for (const auto& kv : map) {
            if (kv.second != 0) {
                return false;
            }
        }
        return true;
    }
};

File: 1556-make-two-arrays-equal-by-reversing-subarrays/1556-make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        HashMap<Integer , Integer> map1 = new HashMap<>();
        HashMap<Integer , Integer> map2 = new HashMap<>();
        for(int i : arr) {
            map1.put(i , map1.getOrDefault(i , 0) + 1);
        }
        for(int i : target) {
            map2.put(i , map2.getOrDefault(i , 0) + 1);
        }
        return map1.equals(map2);
    }
}